### PR TITLE
AC-1428: Moving off of Brightback

### PR DESCRIPTION
**Pull request by shrirupa**
AC-1428-Moving off of Brightback

### What Changed
 - Removed all brightback related code
 - On Account Cancellation, you should see a pendo guide instead of Brightback and if the guide doesn't load you should still be able to cancel account.
- Added error reporting to sentry if cancellation guide fails to load.
- Removed brightback from showing up when downgrading to free plan.

### How To Test
 - Try cancelling account on both app.sparkpost.test:3100 (guide will show up here) and on localhost:3100
- Try downgrading to free plan from a paid plan 
### To Do
- [ ] Address feedback
- [ ] Remember to move the Cancellation Guide to Public in pendo.

### CODE REVIEW

File: src/helpers/errorTracker.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ const BLACKLIST = new Set([
   '@@redux-form/SET_SUBMIT_FAILED',
 ]);
 
+/*report that guide couldn't be loaded */
+export function reportGuideLoadingError(err) {
```
Comment (bkemper) [inline src/helpers/errorTracker.js:26]: Please use the `report` method.

```
import ErrorTracker from 'src/helpers/errorTracker';
ErrorTracker.report('pendo-guide', error);
```

Comment (shrirupa) [reply src/helpers/errorTracker.js:26]: Thanks!

File: src/pages/account/components/CancellationPanel.js [hunk 2]
```diff
@@ -21,6 +20,17 @@ export class CancellationPanel extends React.Component {
...
   };
 
+  handleCancelAccount = () => {
+    try {
+      if (!window.pendo.showGuideById(GUIDE_IDS.CANCEL_ACCOUNT)) {
+        this.props.history.push(ACCOUNT_CANCEL_LINK);
+      }
+    } catch (error) {
+      ErrorTracker.report('account-cancellation-pendo-guide-error', error);
+      this.props.history.push(ACCOUNT_CANCEL_LINK);
```
Comment (bkemper) [inline src/pages/account/components/CancellationPanel.js:30]: We should change this experience to show our own confirmation modal.

Comment (shrirupa) [reply src/pages/account/components/CancellationPanel.js]: Done 👍. I have added the content of this fallback Modal to same as pendo Guide 'Important' section.

File: src/pages/account/components/CancellationPanel.js [hunk 3]
```diff
@@ -1,54 +1,73 @@
...
-import config from 'src/config';
-import Brightback from 'src/components/brightback/Brightback';
+import { GUIDE_IDS } from 'src/constants';
 import { formatDate } from 'src/helpers/date';
-
+import { withRouter } from 'react-router-dom';
+import ErrorTracker from 'src/helpers/errorTracker';
 const ACCOUNT_CANCEL_LINK = '/account/cancel';
 
-export class CancellationPanel extends React.Component {
```
Comment (bkemper) [inline src/pages/account/components/CancellationPanel.js:15]: 👏

File: src/pages/account/components/CancellationPanel.js [hunk 4]
```diff
@@ -1,54 +1,73 @@
...
       return fetchAccount();
     });
   };
 
-  render() {
-    const { account } = this.props;
-    const { pending_cancellation, cancelLoading } = account;
+  const handleCancelAccount = () => {
+    try {
+      if (!window.pendo.showGuideById(GUIDE_IDS.CANCEL_ACCOUNT)) {
```
Comment (bkemper) [inline src/pages/account/components/CancellationPanel.js:26]: This try/catch logic should probably be wrapped up in a helper.

File: src/pages/account/components/CancellationPanel.js [hunk 5]
```diff
@@ -80,29 +99,35 @@ export class CancellationPanel extends React.Component {
...
       </Panel>
-    );
-  }
+      <ConfirmationModal
+        open={showCancelAccountModal}
+        title="Are you sure you want to cancel your account?"
+        content={
+          <p>
+            If you cancel, you can send emails up to your monthly limit before your cancel date, but
+            you cannot overage past that. You will not be able to validate email addresses using
```
Comment (bkemper) [inline src/pages/account/components/CancellationPanel.js:113]: 👍

File: src/pages/account/components/CancellationPanel.js [hunk 6]
```diff
@@ -1,54 +1,73 @@
...
-
+import { withRouter } from 'react-router-dom';
+import ErrorTracker from 'src/helpers/errorTracker';
 const ACCOUNT_CANCEL_LINK = '/account/cancel';
 
-export class CancellationPanel extends React.Component {
-  onRenewAccount = () => {
-    const { renewAccount, fetchAccount, showAlert } = this.props;
+export function CancellationPanel(props) {
+  const [showCancelAccountModal, setShowCancelAccountModal] = useState(false);
```
Comment (bkemper) [inline src/pages/account/components/CancellationPanel.js:15]: You don't really need it, but for future reference, I created a `useModal` hook.

File: src/pages/account/components/CancellationPanel.js [hunk 7]
```diff
@@ -1,54 +1,73 @@
...
 
-    if (pending_cancellation) {
-      return (
-        <Panel sectioned title="Pending Account Cancellation">
-          <Stack>
-            <Heading as="h3" looksLike="h6">
-              Account is set to cancel {formatDate(pending_cancellation.effective_date)}
-            </Heading>
+  const handleCancel = () => {
+    props.history.push(ACCOUNT_CANCEL_LINK);
```
Comment (bkemper) [inline src/pages/account/components/CancellationPanel.js:36]: I know we can't change this now, but this scares me that we have a landing page that will take immediate action.

File: src/pages/billing/components/SubmitSection.js [hunk 8]
```diff
@@ -1,30 +1,26 @@
...
-        >
-          Change Plan
-        </Button>
-      )}
-    />
+    <Button
+      type={isDowngradeToFree ? 'button' : 'submit'}
+      to={
+        isDowngradeToFree
+          ? `/account/billing/plan/change?immediatePlanChange=${freePlan.code}`
```
Comment (bkemper) [inline src/pages/billing/components/SubmitSection.js:16]: Same feeling about this.

File: src/pages/billing/forms/ChangePlanForm.js [hunk 9]
```diff
@@ -181,7 +183,8 @@ export const ChangePlanForm = ({
                   loading={submitting}
                   selectedBundle={selectedBundle}
                   account={account}
-                  brightbackCondition={isDowngradeToFree}
+                  isDowngradeToFree={isDowngradeToFree}
+                  freePlan={_.find(plans, { isFree: true })}
```
Comment (bkemper) [inline src/pages/billing/forms/ChangePlanForm.js:187]: Do we have a selector for this?

File: src/pages/billing/forms/ChangePlanForm.js [hunk 10]
```diff
@@ -202,6 +205,8 @@ const mapStateToProps = (state, props) => {
     canUpdateBillingInfo: canUpdateBillingInfoSelector(state),
     currentPlan: currentPlanSelector(state),
     promoCodeObj: getPromoCodeObject(state),
+    loading: state.billing.plansLoading,
```
Comment (bkemper) [inline src/pages/billing/forms/ChangePlanForm.js:208]: Is this working as expected?  It looks like `loading` is being provided by `useChangePlanContext`.

**Comment by shrirupa** [comment]
> I was unable to test the Pendo guide. How are be going to test this in staging/production? I left a few comments.

This pendo guide's status is set to `Staged`, so locally it should be able to visible on `http://app.sparkpost.test:3100/account/settings`, since `app.sparkpost.test:3100` is set as one of the staging server in pendo. When this ticket goes to prod I can change the guide status to `public` in pendo.

File: src/pages/account/components/CancellationPanel.js (hunk 7 above)
Comment (shrirupa) [reply src/pages/account/components/CancellationPanel.js:36]: I agree! good thing is atleast this action can be undone very easily, which is not the same with ImmediateChangePlanPage. I am thinking I can change this pattern while downgrading to free.

File: src/pages/billing/components/SubmitSection.js (hunk 8 above)
Comment (shrirupa) [reply src/pages/billing/components/SubmitSection.js]: Removing the ImmediateChangePlanPage helped simplify this.

File: src/pages/billing/forms/ChangePlanForm.js (hunk 9 above)
Comment (shrirupa) [reply src/pages/billing/forms/ChangePlanForm.js]: we don't have a selector for this, but won't be needed with the removal of immediateChanagePlanPage.

File: src/pages/billing/forms/ChangePlanForm.js [hunk 11]
```diff
@@ -29,6 +28,7 @@ import { useChangePlanContext } from '../context/ChangePlanContext';
 import { FeatureChangeContextProvider } from '../context/FeatureChangeContext';
 import OGStyles from './ChangePlanForm.module.scss';
 import HibanaStyles from './ChangePlanFormHibana.module.scss';
+import * as conversions from 'src/helpers/conversionTracking';
```
Comment (bkemper) [inline src/pages/billing/forms/ChangePlanForm.js:31]: We can remove this.

File: src/config/routes/index.js [hunk 12]
```diff
@@ -530,15 +530,6 @@ const routes = [
     supportDocSearch: 'upgrade account',
     category: 'Account',
   },
-  {
-    path: '/account/billing/plan/change',
-    component: billing.ImmediateChangePlanPage,
```
Comment (bkemper) [inline src/config/routes/index.js:535]: 👏

Comment (bkemper) [reply src/config/routes/index.js:535]: Looks like `stripImmediatePlanChange` is no longer being used and can be deleted.

**Pull request merged**